### PR TITLE
Classes- input reader

### PR DESCRIPTION
**Pull request by Rahulkug**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [ ] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

File: solutions/classes/input-reader.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+let idx = 0;
+data = data.split("\n");
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+const Inputreader = class {
```
Comment (shivateja1729) [inline solutions/classes/input-reader.js:12]: [naming conventions ](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions) are not full filled with variable, function and class names.  like `Inputreader`  `Data`  `read_integers` `read_strings`  `typeofinput ` etc.,

File: solutions/classes/input-reader.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  read_floats() {
+    console.log(this.Index + " " + parseFloat(this.Data).toFixed(2));
+  }
+};
+
+var Num = parseInt(readLine());
+var typeofinput = readLine();
+for (let i = 0; i < Num; i++) {
+  if (typeofinput === "string") {
+    new Inputreader(i).read_strings();
```
Comment (shivateja1729) [inline solutions/classes/input-reader.js:32]: I think its better to store the instance of the class as a class variable first, and then use it everytime.
like `let NewInputReader = new InputReader(i)`. Please correct me if its not a good practice of storing as variables.

Comment (shivateja1729) [reply solutions/classes/input-reader.js:32]: > 

@sailesh-ramayanam  should i make the above change in the code?

**Review by shivateja1729 — APPROVED** [review]

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

File: solutions/classes/input-reader.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+for (let i = 0; i < num; i++) {
+	let NewInputReader = new InputReader(i)	
+	if (typeOfInput === "string") {
+		NewInputReader.readStrings();
+	} else if (typeOfInput === "integer") {
+		NewInputReader.readIntegers();
+	} else {
+		NewInputReader.readFloats();
+	}
+}
```
[the remark this answers is not in the document]
Comment (shivateja1729) [reply solutions/classes/input-reader.js]: @sailesh-ramayanam edited the above changes in the code and problem description.

File: solutions/classes/input-reader.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+// -------- Do NOT edit anything above this line ----------
+class InputReader {
+  data = readLine();
+  constructor(Index) {
+    this.Index = Index;
+  }
+  readStrings() {
+    console.log(this.Index + " " + this.data);
```
[the remark this answers is not in the document]
Comment (shivateja1729) [reply solutions/classes/input-reader.js]: @sailesh-ramayanam addressed.

**Comment by sailesh-ramayanam** [comment]
@shivateja1729 Template is missing.

**Comment by shivateja1729** [comment]
> @shivateja1729 Template is missing.

@sailesh-ramayanam  template was not mentioned in problem description. Seems like there is no template for this problem.

**Comment by sailesh-ramayanam** [comment]
@shivateja1729 We should always provide template for `Functions`, `Classes` and `Linked Lists`. Otherwise, students will simply solve with a for loop without using the relevant concept.

**Comment by shivateja1729** [comment]
> @shivateja1729 We should always provide template for `Functions`, `Classes` and `Linked Lists`. Otherwise, students will simply solve with a for loop without using the relevant concept.

@sailesh-ramayanam  added the template and updated the problem statement as per the template.

File: solutions/classes/input-reader.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+class InputReader {
+  data = readLine();
+  constructor(Index) {
+    this.Index = Index;
```
[the remark this answers is not in the document]
Comment (shivateja1729) [reply solutions/classes/input-reader.js]: @sailesh-ramayanam made changes.

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**